lua/nvim_lsp/jdtls.lua

```diff
@@ -0,0 +1,30 @@
+local skeleton = require 'nvim_lsp/skeleton'
+local util = require 'nvim_lsp/util'
+local lsp = vim.lsp
+
+local cwd = vim.loop.cwd()
+
+local server_name = "jdtls"
+local bin_name = "jdtls"
+
+skeleton[server_name] = {
+  default_config = {
+    cmd = {"jdtls"};
+    filetypes = {"java"};
+    root_dir = function() return cwd end;
```

Comment:
this will always return the CWD found when `jdtls.lua` was first evaluated. Is that intentional? Instead this could be:

    root_dir = vim.loop.cmd

then the actual-current directory will be returned, after user does `:cd`.

Reply:
No, I just did not think or it. I like this better indeed. I'll change it in a minute.

Reply:
Ghhm, vim.loop.cmd seems to be nil. Unfortunately, I'm quite foreign to Lua and to vim scripting in general. Unfortunately I do not have the time right now to learn vim scripting right now. So if anybody could either fix it or tell me what should work, that would be great!

Reply:
It's `vim.loop.cwd` as in `w` not `m` :P

@justinmk The pattern of always returning a single value is to enforce that even if the cwd changes, only one instance of the server is spawned. Basically there's 2 major patterns for servers:

1. spawn a new server for each project root
2. start just one server for each neovim session

I was thinking of making this a util function called `util.once(value)` which returns the same value forever.

Reply:
Ah, that makes sense. Again both vim scripting and Lua are foreign to me. I just love Language server and this implementation, so I'll gladly help wherever I can.

Do I understand correctly that maybe it's not the best idea then to change the root dir if the cwd changes?
If this is the case. Do you have any further remarks on the MR? :)

Reply:
Yup that would not be good. So basically just answer the question: how many instances of the server do you want? If the answer is 1 per neovim then keep things as is (and merge origin/master).

Otherwise use something like a `rootPattern` you can see in some other lsp files.

Reply:
One per instance seems good to me. Maybe altering the root pattern might be a good idea to do eventually, but this might be different for different frameworks and build systems. I guess this is a fine start.